modules/core/src/main/scala/iota/internal/toolbelts.scala

```diff
@@ -163,4 +223,30 @@ private[internal] sealed abstract class IotaCommonToolbelt {
       case t => Left(s"unexpected type $t ${showRaw(t)} when destructuring CopK $tpe")
     }
 
+  class TypeListBuilder private[IotaCommonToolbelt](
+    consTpe: Type,
+    nilTpe: Type
+  ) {
+    final def apply(tpes: List[Type]): Type =
+      tpes.reverse.foldLeft(nilTpe)((t, h) => makeCons(h, t))
+
+    private[this] lazy val (consPrefix, consSym) = consTpe match {
+      case TypeRef(prefix, sym, _) => (prefix, sym)
+      case _ => sys.error("internal iota initialization error")
```

Comment:
I couldn't figure out a way to extract the prefix of the type without this ugly match expression (or unusual casting).